### ISSUE TITLE
Invalid block warning dropdown menu cannot be navigated using arrow keys.

### ISSUE THREAD

**Issue by talldan** [WordPress/gutenberg#24303]
**Describe the bug**
When a user has an invalid block, the extra options in the dropdown cannot be navigated using a keyboard. The dropdown closes and focus is moved in an unexpected way.

Possibly another issue with `WritingFlow` interrupting keyboard navigation.

**To reproduce**
1. Add a block and make it invalid by switching to the code editor and editing the markup (e.g. changing the tags)
2. Try opening the additional options using the keyboard and navigating between the options using arrow keys.
3. Observe the dropdown menu closes unexpectedly and focus jumps to elsewhere in the editor

**Expected behavior**
The user can navigate between the options and the menu stays open.

**Screenshots**
<img width="465" alt="Screenshot 2020-07-31 at 1 56 36 pm" src="https://user-images.githubusercontent.com/677833/89004851-aef15380-d335-11ea-9a11-c60655fe583e.png">
